src/nvim/api/window.c

```diff
@@ -462,3 +464,104 @@ void nvim_win_set_hl_ns(Window window, Integer ns_id, Error *err)
   win->w_hl_needs_update = true;
   redraw_later(win, UPD_NOT_VALID);
 }
+
+/// Computes the number of screen lines occupied by a range of text in a given window.
+/// Works for off-screen text and takes folds into account.
+///
+/// Diff filler or virtual lines above a line are counted as a part of that line,
+/// unless the line is on "start_row" and "start_vcol" is specified.
+///
+/// Diff filler or virtual lines below the last buffer line are counted in the result
+/// when "end_row" is omitted.
+///
+/// Line indexing is similar to |nvim_buf_get_text()|.
+///
+/// @param window  Window handle, or 0 for current window.
+/// @param opts    Optional parameters:
+///                - start_row: Starting line index, 0-based inclusive.
+///                             When omitted start at the very top.
+///                - end_row: Ending line index, 0-based inclusive.
+///                           When omitted end at the very bottom.
+///                - start_vcol: Starting virtual column index on "start_row",
+///                              0-based inclusive, rounded down to full screen lines.
+///                              When omitted include the whole line.
+///                - end_vcol: Ending virtual column index on "end_row",
+///                            0-based exclusive, rounded up to full screen lines.
+///                            When omitted include the whole line.
+/// @return  The number of screen lines that the range of text occupy.
+///
+/// @see |virtcol()| for text width.
+Object nvim_win_text_height(Window window, Dict(win_text_height) *opts, Error *err)
```

Comment:
Thinking about this more, I don't think Object helps us with forward-compatibility. Although technically we might not violate `:help api-contract` because we won't need to change the formal signature, if we return a new kind of result, it will still break clients expecting to "unbox" an int.

And given the `..._height` suffix of the function, potential for other kinds of results is limited. So I guess this return type should be `Integer`.

Reply:
I was thinking that in future there may be an option (e.g. `details`) that allows returning different kinds of heights, which will require a different return type. Of course it's also possible to return only one integer in each call with a different option.

Reply:
Oh, do you mean this may break client wrappers?

Reply:
> Oh, do you mean this may break client wrappers?

Yes, especially statically-compiled languages. Makes auto-generation harder.



> may be an option (e.g. `details`) that allows returning different kinds of heights,

In that case we could change this to `Array(Integer)` ? That avoids needing to worry about key names. But I also think `Dict(Integer)` with key name `height` would be fine for now.